x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/UserBoolQueryBuilder.java

```diff
@@ -94,8 +49,9 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) throws
         return super.doRewrite(queryRewriteContext);
     }
 
-    boolean isIndexFieldNameAllowed(String queryFieldName) {
-        // Type is needed to filter on user doc type
-        return queryFieldName.equals("type") || USER_FIELD_NAME_TRANSLATOR.supportedIndexFieldName(queryFieldName);
+    boolean isIndexFieldNameAllowed(String fieldName) {
```

Comment:
nit: same as in the api query builder, the additional fields could be defined in a static set in the class.

Reply:
Pushed [bde986f](https://github.com/elastic/elasticsearch/pull/109559/commits/bde986f27aa7e06e7211c2d9df47a60aab0e1e7b)